src/plugins/pol/polformat.cpp

```diff
@@ -20,320 +20,236 @@
-            stringData[1]     = (uint8_t)((uint8_t *) (&uint64Entry->data))[1];
-            stringData[2]     = (uint8_t)((uint8_t *) (&uint64Entry->data))[2];
-            stringData[3]     = (uint8_t)((uint8_t *) (&uint64Entry->data))[3];
-            stringData[4]     = (uint8_t)((uint8_t *) (&uint64Entry->data))[4];
-            stringData[5]     = (uint8_t)((uint8_t *) (&uint64Entry->data))[5];
-            stringData[6]     = (uint8_t)((uint8_t *) (&uint64Entry->data))[6];
-            stringData[7]     = (uint8_t)((uint8_t *) (&uint64Entry->data))[7];
-            result.data       = stringData;
+            auto &data = registryEntry->data =
+                    QString::fromStdString(std::get<std::string>(entry.data));
+
+            return registryEntry;
         }
-        break;
-        case REG_MULTI_SZ: {
-            auto binaryEntry = static_cast<RegistryEntry<QStringList> *>(entry.get());
-            QByteArray byteArray;
-            for (const QString &str : binaryEntry->data)
-            {
-                auto sixteenBitString = str.toStdU16String();
-                size_t size           = sixteenBitString.size() * sizeof(char16_t);
-                byteArray.append(reinterpret_cast<const char *>(sixteenBitString.c_str()), size);
-                byteArray.append(2, '\0');
-            }
-            if (byteArray.size() == 0)
-            {
-                byteArray.append(4, '\0');
-            }
-            else
-            {
-                byteArray.append(2, '\0');
+        case pol::PolicyRegType::REG_MULTI_SZ: {
+            auto registryEntry = std::make_unique<model::registry::RegistryEntry<QStringList>>();
+
+            registryEntry->key = QString::fromStdString(key);
+            registryEntry->value = QString::fromStdString(value);
+            registryEntry->type = model::registry::RegistryEntryType::REG_MULTI_SZ;
+
+            auto &data = std::get<std::vector<std::string>>(entry.data);
+            registryEntry->data.reserve(data.size());
+            for (const auto &datum : data) {
+                registryEntry->data.append(QString::fromStdString(datum));
             }
-            char *stringData = new char[byteArray.size()];
-            memcpy(stringData, byteArray, byteArray.size());
-            result.data = stringData;
-            result.size = byteArray.size();
+
+            return registryEntry;
+        }
+        case pol::PolicyRegType::REG_QWORD_LITTLE_ENDIAN: {
+            auto registryEntry = std::make_unique<model::registry::RegistryEntry<uint64_t>>();
+
+            registryEntry->key = QString::fromStdString(key);
+            registryEntry->value = QString::fromStdString(value);
+            registryEntry->type = model::registry::RegistryEntryType::REG_QWORD;
+            registryEntry->data = std::get<uint64_t>(entry.data);
+
+            return registryEntry;
         }
-        break;
+        case pol::PolicyRegType::REG_QWORD_BIG_ENDIAN:
```

Comment:
Why we do not have implementation of `REG_QWORD_BIG_ENDIAN` it must be similar in function to `REG_QWORD_LITTLE_ENDIAN`.